miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultBoxService.java

```diff
@@ -0,0 +1,282 @@
+
+  @Autowired
+  private NamingScheme namingScheme;
+
+  @Autowired
+  private SecurityProfileStore securityProfileStore;
+
+  private void addBoxContentsChangeLog(Box box, String message) throws IOException {
+    BoxChangeLog changeLog = new BoxChangeLog();
+    changeLog.setBox(box);
+    changeLog.setTime(new Date());
+    changeLog.setColumnsChanged("contents");
+    changeLog.setUser(authorizationManager.getCurrentUser());
+    changeLog.setSummary(message);
+    changeLogStore.create(changeLog);
+  }
+
+  private void applyChanges(Box from, Box to) throws IOException {
+    to.setAlias(from.getAlias());
+    to.setDescription(from.getDescription());
+    to.setIdentificationBarcode(from.getIdentificationBarcode());
+    to.setUse(boxStore.getUseById(from.getUse().getId()));
+  }
+
+  @Override
+  public long count(Consumer<String> errorHandler, PaginationFilter... filter) throws IOException {
+    return boxStore.count(errorHandler, filter);
+  }
+
+  @Override
+  public void deleteBox(Box box) throws IOException {
+    authorizationManager.throwIfNotWritable(box);
+    if (!boxStore.remove(box)) {
+      throw new IOException("Unable to delete box.");
+    }
+  }
+
+  @Override
+  public void discardAllTubes(Box box) throws IOException {
+    authorizationManager.throwIfNotWritable(box);
+
+    boxStore.discardAllTubes(box);
+  }
+
+  @Override
+  public void discardSingleTube(Box box, String position) throws IOException {
+    authorizationManager.throwIfNotWritable(box);
+    boxStore.discardSingleTube(box, position);
+  }
+
+  @Override
+  public Box get(long boxId) throws IOException {
+    Box o = boxStore.get(boxId);
+    if (authorizationManager.readCheck(o)) return o;
+    else throw new IOException("User " + authorizationManager.getCurrentUser().getFullName() + " cannot read Box " + boxId);
+
+  }
+
+  @Override
+  public AuthorizationManager getAuthorizationManager() {
+    return authorizationManager;
+  }
+
+  @Override
+  public PaginatedDataSource<Box> getBackingPaginationSource() {
+    return boxStore;
+  }
+
+  @Override
+  public Box getByAlias(String alias) throws IOException {
+    Box o = boxStore.getBoxByAlias(alias);
+    if (authorizationManager.readCheck(o)) return o;
+    else throw new IOException("User " + authorizationManager.getCurrentUser().getFullName() + " cannot read Box " + alias);
+
+  }
+
+  @Override
+  public Box getByBarcode(String barcode) throws IOException {
+    Box o = boxStore.getByBarcode(barcode);
+    if (authorizationManager.readCheck(o)) return o;
+    else throw new IOException("User " + authorizationManager.getCurrentUser().getFullName() + " cannot read Box " + barcode);
+  }
+
+  @Override
+  public Map<String, Integer> getColumnSizes() throws IOException {
+    if (boxStore != null) {
+      return boxStore.getBoxColumnSizes();
+    } else {
+      throw new IOException("No boxStore available. Check that it has been declared in the Spring config.");
```

Comment:
Sacre bleu, what is thees? How on earth could you mees such a sweet little succulent RequestManager holdover? 
https://giphy.com/gifs/hannibal-nbc-su-zakana-3m29WMD6BNss